Cargo.toml

```diff
@@ -14,20 +14,21 @@ default-run = "dft"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-arrow-flight = { version = "52.2.0", features = ["flight-sql-experimental"] , optional = true }
+arrow-flight = { version = "53.0.0", features = ["flight-sql-experimental"] , optional = true }
 async-trait = "0.1.80"
 clap = { version = "4.5.1", features = ["derive"] }
 color-eyre = "0.6.3"
 crossterm = { version = "0.28.1", features = ["event-stream"] }
-datafusion = "41.0.0"
-datafusion-common = "41.0.0"
+datafusion = "42.0.0"
+datafusion-common = "42.0.0"
 datafusion-functions-json = { version = "0.41.0", optional = true }
 deltalake = { version = "0.19.0", features = ["datafusion"], optional = true }
 directories = "5.0.1"
 env_logger = "0.11.5"
 futures = "0.3.30"
 http = "0.2.9"
 http-body = {version = "0.4.5" }
+hudi = { features = ["datafusion"], optional = true, git = "https://github.com/apache/hudi-rs", rev = "abb3e635a28af229ba1e872fd6005c381e2b7ded" }
```

Comment:
you can use https://crates.io/crates/hudi/0.2.0-rc.1 now, and soon use `0.2.0` once the release is out.

Reply:
@matthewmturner thanks for the work - any plan to move this forward?

Reply:
Yes! I'm really looking forward to getting this merged. Unfortunately, it was held back back because we already have a deltalake feature but they [werent able to upgrade datafusion](https://github.com/delta-io/delta-rs/pull/2886) for an extended period of time which prevented this from moving forward.  It seems to finally be resolved and I hope to get this merged shortly.

Some other context:

1. I dont plan on releasing next version until this and iceberg are added as features.
2. Because of how unfortunate this way i am considering a switch to the new [FFI](https://github.com/apache/datafusion/pull/12920) interface.  The community has [expressed interest](https://github.com/apache/datafusion/issues/13274#issuecomment-2462559814) in this as well, but i wouldnt expect that to come until after a release

Reply:
that's good to know. thanks!

Reply:
@xushiyan do you think https://github.com/apache/hudi-rs/issues/196 could be included in the 0.2.0 release?

Reply:
@matthewmturner 0.2.0 was already in voting stage. we can get this in 0.3.0 which shouldn't be too far away